### ISSUE TITLE
Turning off all rules and then enabling desired rules - Take II

### ISSUE THREAD

**Issue by JamesNK** [dotnet/roslyn-analyzers#2382]
I created https://github.com/dotnet/roslyn/issues/35164, but I believe there is a separate bug.

#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Repro steps

1. Open [Class1.zip](https://github.com/dotnet/roslyn-analyzers/files/3100588/Class1.zip) (has copy of rules from `AllRulesDisabled.ruleset`)
1. Compile

#### Expected behavior

No code analysis warnings because they're all disabled

#### Actual behavior

Code analysis warnings, e.g. warning CA2208: Call the ArgumentOutOfRangeException constructor that contains a message and/or paramName parameter.

```
Microsoft (R) Build Engine version 16.1.65-preview+gc86abee47b for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 171.86 ms for C:\Users\James\source\repos\ClassLibrary7\ClassLibrary7\ClassLibrary7.csproj.
C:\Program Files\dotnet\sdk\3.0.100-preview5-011367\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(151,5): message NETSDK1057: You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview [C:\Users\James\source\repos\ClassLibrary7\ClassLibrary7\ClassLibrary7.csproj]
Class1.cs(9,19): warning CA2208: Call the ArgumentOutOfRangeException constructor that contains a message and/or paramName parameter. [C:\Users\James\source\repos\ClassLibrary7\ClassLibrary7\ClassLibrary7.csproj]
  ClassLibrary7 -> C:\Users\James\source\repos\ClassLibrary7\ClassLibrary7\bin\Debug\netstandard2.0\ClassLibrary7.dll

Build succeeded.

Class1.cs(9,19): warning CA2208: Call the ArgumentOutOfRangeException constructor that contains a message and/or paramName parameter. [C:\Users\James\source\repos\ClassLibrary7\ClassLibrary7\ClassLibrary7.csproj]
    1 Warning(s)
    0 Error(s)
```

![image](https://user-images.githubusercontent.com/303201/56462093-2507ca80-6411-11e9-8311-c0b2c7036468.png)

CA2208 isn't in `AllRulesDisabled.ruleset`.

**Comment by mavasani** [comment]
Are you sure you don’t have a stale copy of AllRulesDisabled.ruleset. I did check yesterday that the ruleset along with 2.9.2 FxCop analyzers package had a entry turning off CA2208.

**Comment by JamesNK** [comment]
I am copying it directly from the NuGet package.

![image](https://user-images.githubusercontent.com/303201/56465346-c9a8fd00-644f-11e9-89d4-55ed551a6b15.png)

**Comment by mavasani** [comment]
@JamesNK - I see that you are using the ruleset from Microsoft.CodeQuality.Analyzers NuGet package, which only has subset of FxCop rules (CA2208 is implemented in Microsoft.NetCore.Analyzers). I would recommend using the rulesets from Microsoft.CodeAnalysis.FxCopAnalyzers NuGet package, which does include entry for this rule.

**Comment by JamesNK** [comment]
Ah ok. I misunderstood the documentation at https://docs.microsoft.com/en-us/visualstudio/code-quality/analyzer-rule-sets?view=vs-2019#use-analyzer-rule-sets

I've submitted a PR to slightly expand the example - https://github.com/MicrosoftDocs/visualstudio-docs/pull/3125

**Comment by mavasani** [comment]
Thanks, resolving this as external
